building.rb

```diff
@@ -1,2 +1,29 @@
 class Building
+	attr_accessor :address, :apartments
+
+	def initialize(args={})
+
+		@address = args[:address]
+		@apartments = args[:apartments]
+	end
+
+	def total_room_count
+		@apartments.map {|apartment| apartment.room_count}.reduce(:+)
+	end
+
+	def total_monthly_revenue
+		@apartments.map {|apartment| apartment.monthly_rent}.reduce(:+)
+	end
+
+	def apartments_by_rent
+		@apartments.sort_by {|apartment| apartment.monthly_rent}.reverse
+	end
+
+	def find_apartments_by_bedroom_count(bedroom_count)
+		@apartments.select {|apartment| return apartment if apartment.bedroom_count == bedroom_count}
```

Comment:
The way you are using select here is actually exactly how .find works. You are calling select but returning as soon as the first match is found, which is what .find does.
```
@apartments.find {|apartment| apartment.bedroom_count == bedroom_count}
```
@wvicioso